src/gui/qgsquerybuilder.cpp

```diff
@@ -94,6 +96,48 @@ QgsQueryBuilder::QgsQueryBuilder( QgsVectorLayer *layer,
   layerSubsetStringChanged();
 
   lblDataUri->setText( tr( "Set provider filter on %1" ).arg( layer->name() ) );
+
+  lblProviderStorageType->setOpenExternalLinks(true);
+
+  //TODO what to put there for other providers, e.g. postgis?
+  QStringList supportedSqlDialectsText = QStringList();
+  if ( layer->dataProvider()->name() == "ogr" ) {
+
+    GDALDriverH driver = GDALGetDriverByName( layer->dataProvider()->storageType().toLocal8Bit().constData() );
+    const char* supportedSqlDialects = GDALGetMetadataItem( driver, GDAL_DMD_SUPPORTED_SQL_DIALECTS, nullptr );
+
+    QStringList supportedSqlDialectsList = QString(supportedSqlDialects).split(" ");
+    int count = supportedSqlDialectsList.count();
+    for( int i=0; i<count; i++ )
+    {
+      QString supportedSqlDialect = supportedSqlDialectsList.at(i);
+      QString link = QString();
+      QString title = supportedSqlDialect;
+      if ( supportedSqlDialect == "NATIVE" ) {
+        const QString gdalDriverHelpTopic = GDALGetMetadataItem( driver, GDAL_DMD_HELPTOPIC, nullptr );  // e.g. "drivers/vector/ili.html"
+        link = "https://gdal.org/en/latest/" + gdalDriverHelpTopic;
```

Comment:
I would suggest to create a small  helper function in src/core/qgsgdalutils.cpp like:
```
static QString gdalDocumentationUrlForDriver(GDALDriverH hDriver);
```
It might be prudent to be robust to the GDALGetMetadataItem() returning nullptr, and in that case to return an empty string. Normally almost all in-tree GDAL drivers have one, but there might a couple esoteric ones which don't, and for out-of-tree drivers.

It could be used by helpCreationOptionsFormat() in the same file. And I suspect there are a number of other places where pointing to GDAL driver doc could be useful.